udata/tests/api/test_me_api.py

```diff
@@ -307,3 +315,100 @@ def test_clear_apikey(self):
 
         self.user.reload()
         self.assertIsNone(self.user.apikey)
+
+    def test_delete(self):
+        '''It should delete the connected user'''
+        user = self.login()
+        self.assertIsNone(user.deleted)
+        other_user = UserFactory()
+        members = [Member(user=user), Member(user=other_user)]
+        organization = OrganizationFactory(members=members)
+        disc_msg_content = faker.sentence()
+        disc_msg = DiscMsg(content=disc_msg_content,
+                           posted_by=user)
+        other_disc_msg_content = faker.sentence()
+        other_disc_msg = DiscMsg(content=other_disc_msg_content,
+                                 posted_by=other_user)
+        discussion = DiscussionFactory(user=user,
+                                       discussion=[disc_msg, other_disc_msg])
+        issue_msg_content = faker.sentence()
+        issue_msg = IssueMsg(content=issue_msg_content,
+                             posted_by=user)
+        other_issue_msg_content = faker.sentence()
+        other_issue_msg = IssueMsg(content=other_issue_msg_content,
+                                   posted_by=other_user)
+        issue = IssueFactory(user=user,
+                             discussion=[issue_msg, other_issue_msg])
+        dataset = DatasetFactory(owner=user)
+        reuse = ReuseFactory(owner=user)
+        resource = CommunityResourceFactory(owner=user)
+        activity = UserCreatedDatasetFactory(actor=user, related_to=dataset)
+
+        following = FollowFactory(follower=user, following=other_user)
+        followed = FollowFactory(follower=other_user, following=user)
+
+        with self.capture_mails() as mails:
+            response = self.delete(url_for('api.me'))
+        self.assertEqual(len(mails), 1)
+        self.assertEqual(mails[0].send_to, set([user.email]))
+        self.assertEqual(mails[0].subject, 'Account deletion')
+        self.assert204(response)
+
+        user.reload()
+        organization.reload()
+        discussion.reload()
+        issue.reload()
+        dataset.reload()
+        reuse.reload()
+        resource.reload()
+        activity.reload()
+
+        # The following are deleted
+        with self.assertRaises(Follow.DoesNotExist):
+            following.reload()
+        # The followers are deleted
+        with self.assertRaises(Follow.DoesNotExist):
+            followed.reload()
+
+        # The personal data of the user are anonymized
+        self.assertEqual(user.email, 'Deleted-{}'.format(user.id))
```

Comment:
It might be safer to have something that validates as a mail (ex: `{user.id}@deleted`).